src/libraries/System.Runtime.InteropServices.JavaScript/tests/System.Runtime.InteropServices.JavaScript.UnitTests/System/Runtime/InteropServices/JavaScript/JSExportTest.cs

```diff
@@ -24,6 +24,17 @@ public async Task JsExportBooleanAsync(bool value)
                 "boolean");
         }
 
+        [Theory]
+        [MemberData(nameof(MarshalInt32Cases))]
+        public async Task JsExportInt32OneWay(int value)
+        {
+            JavaScriptTestHelper.optimizedReached=0;
+            
+            JavaScriptTestHelper.invoke1O(value);
+            await Task.Yield();
```

Comment:
Is Task.Yield guaranteed to empty the work queue? This looks like it could get flaky and might need to be a loop with an iteration limit, since the test itself is async

Reply:
I replaced it with JS side `delay()` which is `setTimer` based. That will be better because across threads, local `Task.Yield` would not help.